scripts/west_commands/runners/jlink.py

```diff
@@ -262,14 +262,14 @@ def flash(self, **kwargs):
         # Get the build artifact to flash, preferring .hex over .bin
         if self.hex_name is not None and os.path.isfile(self.hex_name):
             flash_file = self.hex_name
-            flash_cmd = f'loadfile {self.hex_name}'
+            flash_cmd = 'loadfile {}'.format('"%s"' % self.hex_name)
```

Comment:
Is this not possible with f-strings?

Reply:
```suggestion
            flash_cmd = f'loadfile "{hex_name}"'
```

Reply:
Unfortunately it does not work under Arch Linux.

<pre>
 ✘  ~/P/C/zephyr  west flash -d build --skip-rebuild --hex-file build/firmware/factory\ \(copy\).hex
-- west flash: using runner jlink
-- runners.jlink: JLink version: 7.54b
FATAL ERROR: Cannot flash; no hex (build/firmware/factory (copy).hex) or bin (build/zephyr/zephyr.bin) files found.
An exception trace has been saved in /tmp/west-exc-cejcw9rz.txt
</pre>

<pre>
 ✘  ~/P/C/zephyr  cat /tmp/west-exc-cejcw9rz.txt
Traceback (most recent call last):
  File "/home/bartekk//zephyr/scripts/west_commands/run_common.py", line 228, in do_run_common
    runner.run(command_name)
  File "/home/bartekk/zephyr/scripts/west_commands/runners/core.py", line 537, in run
    self.do_run(command, **kwargs)
  File "/home/bartekk/zephyr/scripts/west_commands/runners/jlink.py", line 224, in do_run
    self.flash(**kwargs)
  File "/home/bartekk/zephyr/scripts/west_commands/runners/jlink.py", line 278, in flash
    raise ValueError(err.format(self.hex_name, self.bin_name))
ValueError: Cannot flash; no hex (build/firmware/factory (copy).hex) or bin (build/zephyr/zephyr.bin) files found.
</pre>

No difference between hex and bin files.

Reply:
It works like a charm :) Thanks!